src/api/Authentication/DatabaseAuthenticationClaimsTransformation.cs

```diff
@@ -21,26 +22,45 @@ public DatabaseAuthenticationClaimsTransformation(BdmsContext dbContext)
     /// <inheritdoc/>
     public async Task<ClaimsPrincipal> TransformAsync(ClaimsPrincipal principal)
     {
-        var userId = principal.Claims.FirstOrDefault(c => c.Type == ClaimTypes.NameIdentifier);
-        if (userId is null) return principal;
-
-        var authenticatedUser =
-            dbContext.Users.FirstOrDefault(u => u.SubjectId == userId.Value) ?? new User { SubjectId = userId.Value };
-
-        if (authenticatedUser.IsDisabled) return principal;
-
-        authenticatedUser.FirstName = principal.Claims.FirstOrDefault(c => c.Type == ClaimTypes.GivenName)?.Value ?? authenticatedUser.FirstName;
-        authenticatedUser.LastName = principal.Claims.FirstOrDefault(c => c.Type == ClaimTypes.Surname)?.Value ?? authenticatedUser.LastName;
-        authenticatedUser.Name = $"{authenticatedUser.FirstName[0]}. {authenticatedUser.LastName}";
-        dbContext.Update(authenticatedUser);
-        await dbContext.SaveChangesAsync().ConfigureAwait(false);
+        var authenticatedUser = await GetOrInsertUser(principal).ConfigureAwait(false);
+        if (authenticatedUser is null)
+            return principal;
 
         var claimsIdentity = new ClaimsIdentity();
         claimsIdentity.AddClaim(new Claim(ClaimTypes.Name, authenticatedUser.Name));
-        if (authenticatedUser.IsAdmin) claimsIdentity.AddClaim(new Claim(ClaimTypes.Role, PolicyNames.Admin));
-        else if (authenticatedUser.IsViewer) claimsIdentity.AddClaim(new Claim(ClaimTypes.Role, PolicyNames.Viewer));
-
+        if (authenticatedUser.IsAdmin)
+            claimsIdentity.AddClaim(new Claim(ClaimTypes.Role, PolicyNames.Admin));
+        else if (authenticatedUser.IsViewer)
+            claimsIdentity.AddClaim(new Claim(ClaimTypes.Role, PolicyNames.Viewer));
         principal.AddIdentity(claimsIdentity);
```

Comment:
PP: Die `if`-Anweisung in geschweifte Klammern packen oder mit mehr Whitespace arbeiten, da aktuell der Code _unruhig_ aussieht. zB
```csharp
        if (authenticatedUser.IsAdmin)
            claimsIdentity.AddClaim(new Claim(ClaimTypes.Role, PolicyNames.Admin));
        else if (authenticatedUser.IsViewer)
            claimsIdentity.AddClaim(new Claim(ClaimTypes.Role, PolicyNames.Viewer));

        principal.AddIdentity(claimsIdentity);
```

oder 
```csharp
        if (authenticatedUser.IsAdmin)
        {
            claimsIdentity.AddClaim(new Claim(ClaimTypes.Role, PolicyNames.Admin));
        }
        else if (authenticatedUser.IsViewer)
        {
            claimsIdentity.AddClaim(new Claim(ClaimTypes.Role, PolicyNames.Viewer));
        }

        principal.AddIdentity(claimsIdentity);
```